components/lib/crash/src/main/java/mozilla/components/lib/crash/service/GleanCrashReporterService.kt

```diff
@@ -0,0 +1,93 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.lib.crash.service
+
+import android.content.Context
+import androidx.annotation.VisibleForTesting
+import mozilla.components.lib.crash.Crash
+import mozilla.components.lib.crash.GleanMetrics.CrashMetrics
+import mozilla.components.support.base.log.logger.Logger
+import mozilla.components.support.ktx.android.content.isMainProcess
+import java.io.File
+
+/**
+ * A [CrashReporterService] implementation for recording metrics with Glean.  The purpose of this
+ * crash reporter is to collect crash count metrics by capturing [Crash.UncaughtExceptionCrash] and
+ * [Crash.NativeCodeCrash] events and record to the respective
+ * [mozilla.components.service.glean.private.CounterMetricType].
+ */
+class GleanCrashReporterService(val context: Context) : CrashReporterService {
+    companion object {
+        // These keys correspond to the labels found for crashCount metric in metrics.yaml
+        const val CRASH_FILE_NAME = "glean_crash_counts"
+        const val UNCAUGHT_EXCEPTION_KEY = "uncaught_exception"
+        const val NATIVE_CODE_CRASH_KEY = "native_code_crash"
+    }
+
+    private val logger = Logger("glean/GleanCrashReporterService")
+
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    internal val file = File(context.filesDir, CRASH_FILE_NAME)
+
+    init {
+        // We only want to record things on the main process because that is the only one in which
+        // Glean is properly initialized.  Checking to see if we are on the main process here will
+        // prevent the situation that arises because the integrating app's Application will be
+        // re-created when prompting to report the crash, and Glean is not initialized there since
+        // it's not technically the main process.
+        if (context.isMainProcess() && file.exists() && file.canRead()) {
+            // Because lib-crash runs in a separate process when report() is called, we need to do
+            // actual recording here because the GleanCrashReporterService constructor is called and
+            // runs in the process where Glean is actually initialized.
+            //
+            // NOTE:  It is important to make sure to initialize Glean before instantiating and
+            // registering the CrashReporter!
```

Comment:
The CrashReporter or the GleanCrashReporterService? 

Also: is it possible to assert this?

Also also: if it is this important we should probably make a note of it in libcrash's integration doc

Reply:
Yeah -- I think there is a use case of an app using `lib-crash` but not `Glean`.  I guess in that case Glean isn't initialized and nothing is recorded, no harm, no foul?  (Since we don't have that use case in Fenix and time is short, maybe just file a bug to investigate that?)

Reply:
Added the Note to the README.md